vars/fcosKolaTestIso.groovy

```diff
@@ -0,0 +1,28 @@
+// Run kola testiso
+// Available parameters:
+//     cosaDir:        string  -- cosa working directory
+//     extraArgs:      string  -- extra arguments to pass to `kola testiso`
+//     skipMetal4k:    boolean -- skip metal4k image
+def call(params = [:]) {
+    stage("Test Live Images") {
+        def cosaDir = utils.getCosaDir(params)
+        def extraArgs = params.get('extraArgs', "");
+
+        try {
+            parallel metal: {
+                shwrap("cd ${cosaDir}  && env TMPDIR=\$(pwd)/tmp/ kola testiso -S ${extraArgs} --output-dir tmp/kola-testiso-metal")
+            }
+            if (!params['skipMetal4k']) {
+                metal4k: {
+                    shwrap("cd ${cosaDir} && env TMPDIR=\$(pwd)/tmp/ kola testiso -S ${extraArgs} --output-dir tmp/kola-testiso-metal4k")
+                }
+            }
```

Comment:
Hmm, I'm not sure if this works. I think you have to make a map first and conditionally add `metal4k` in that map, and then call `parallel` on the map. See in `fcosKola`.

Reply:
I have tested it in both cases, using and not using the skipMetal4k.
Here is the output for it in Jenkins.

`[Pipeline] parallel
[Pipeline] { (Branch: metal)
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
+ set -xeuo pipefail
++ umask
+ '[' 0022 = 0000 ']'
+ cd /srv
++ pwd
+ env TMPDIR=/srv/tmp/ kola testiso -S --output-dir tmp/kola-testiso-metal
Testing scenarios: [pxe-install iso-offline-install pxe-offline-install]
Successfully tested scenario pxe-install for 32.20201104.dev.0 on bios (metal)
Successfully tested scenario pxe-offline-install for 32.20201104.dev.0 on bios (metal)
Successfully tested scenario iso-offline-install for 32.20201104.dev.0 on bios (metal)
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // parallel
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
+ set -xeuo pipefail
++ umask
+ '[' 0022 = 0000 ']'
+ cd /srv
+ kola testiso -S --qemu-native-4k --no-pxe --output-dir tmp/kola-testiso-metal4k
Testing scenarios: [iso-offline-install]
Successfully tested scenario iso-offline-install for 32.20201104.dev.0 on uefi (metal4k)
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] withEnv`

Anyway, I can make it using map as we have in `fcosKola`

Reply:
Hmm interesting that Jenkins doesn't error out on this. But anyway, I think what's it's actually doing there is just running the `metal` branch on its own and then running the `metal4k` code separately. Normally it would've printed something like this (compare with the output of an `fcosKola` run):

```
[Pipeline] parallel
[Pipeline] { (Branch: metal)
[Pipeline] { (Branch: metal4k)
```

`parallel` takes a map and here we're passing it a map with a single key (`metal`).